File: frontend/src/app/rbac/rbacCompat.ts
import type { CapabilityId, ModulePermissionMap, PermissionLevel } from '../types'

const REDMINE_READ = new Set<CapabilityId>([
  'integraciones.provider.redmine.ver',
])

const REDMINE_EDIT = new Set<CapabilityId>([
  'integraciones.provider.redmine.ver',
  'integraciones.provider.redmine.configurar',
  'integraciones.provider.redmine.test_conexion',
  'integraciones.provider.redmine.gestionar_secretos',
  'integraciones.provider.redmine.reportar',
  'integraciones.provider.redmine.vincular',
])

export function getLegacyCapabilityLevel(permissions: ModulePermissionMap, capabilityId: CapabilityId): PermissionLevel {
  const redmineLevel = permissions.redmine
  if (redmineLevel === 'edit' && REDMINE_EDIT.has(capabilityId)) return 'edit'
  if ((redmineLevel === 'read' || redmineLevel === 'edit') && REDMINE_READ.has(capabilityId)) return 'read'
  return 'none'
}

File: backend/app/rbac_compat.py
from __future__ import annotations


REDMINE_LEGACY_READ = {
    "integraciones.provider.redmine.ver",
}

REDMINE_LEGACY_EDIT = {
    "integraciones.provider.redmine.ver",
    "integraciones.provider.redmine.configurar",
    "integraciones.provider.redmine.test_conexion",
    "integraciones.provider.redmine.gestionar_secretos",
    "integraciones.provider.redmine.reportar",
    "integraciones.provider.redmine.vincular",
}


def legacy_capability_level(module_permissions: dict, capability_id: str) -> str | None:
    redmine_level = (module_permissions or {}).get("redmine")
    if redmine_level == "edit" and capability_id in REDMINE_LEGACY_EDIT:
        return "edit"
    if redmine_level in {"read", "edit"} and capability_id in REDMINE_LEGACY_READ:
        return "read"
    return None
